code/modules/admin/admin.dm

```diff
@@ -4639,6 +4639,28 @@ var/global/noir = 0
 		alert("You cannot perform this action. You must be of a higher administrative rank!", null, null, null, null, null)
 		return
 
+/datum/admins/proc/supplydrop_spawn_obj(var/obj/object as text)
+	SET_ADMIN_CAT(ADMIN_CAT_NONE)
+	set desc="(object path) Spawn an object. But all fancy-like"
+	set name="Spawn-Supplydrop"
+	if(!object)
+		return
+	if (usr.client.holder.level >= LEVEL_PA)
+		var/chosen = get_one_match(object)
+		var/preDropTime = 3 SECONDS
+
+		if (chosen)
+			//var/obj/A = new chosen()
+			//A.set_loc(usr.loc)
+			//heavenly_spawn(A)
+			new/obj/effect/supplymarker/safe(usr.loc, preDropTime, chosen)
```

Comment:
Using `usr.loc` instead of `get_turf(usr)` means that this won't work properly when you are observing people. (Note that spawn-heavenly also has this problem and it's annoying.)

Reply:
I changed it to be as follows
```
			var/turf/T = get_turf(usr)
			new/obj/effect/supplymarker/safe(T, preDropTime, chosen)
			logTheThing("admin", usr, null, "spawned [chosen] at ([showCoords(T.x, T.y, T.z)])")
			logTheThing("diary", usr, null, "spawned [chosen] at ([showCoords(T.x, T.y, T.z, 1)])", "admin")
```